projects/frontend/cicd/extract_shared_rc_version.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -e
+
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+
+# Extracts the latest rc-version for the pipeline id that's passed
+if [ $# -eq 0 ]
+  then
+    echo "ERROR: No argument for pipeline id supplied"
+    exit 3
+fi
+npm view @versatiledatakit/shared versions --json | grep $1-rc | cut -d \" -f 2 | tail -1
```

Comment:
Instead of grep and cut may be we can use jq which is a tool for parsing json: `jq -r '.[] | select(contains("909652143-rc"))'` 

Now imagine that npm view doesn't return the json formatted with new lines .

Reply:
A good argument against this is that npm is a third-party tool and we're adding another third-party tool with jq. The current script uses unix built-ins, so we know they'll always behave the same way. Yes, npm is likely to change, but so is jq, so we're increasing the odds of something changing.

Reply:
Also, not sure where to add `jq` if we decide to use it instead. The pipeline currently uses `versatiledatakit/vdk-cicd-base-gui:2.0.0` and I have no idea where it comes from and where it gets built. 😕

Reply:
I also did not know , but searching git repo shows it's produced by here https://github.com/vmware/versatile-data-kit/blob/a374d42c5e9d5800701d9892ade3edb23ee89317/projects/frontend/cicd/.gitlab-ci.yml#L196

If you can address it with sed, that's also fine. You can decide to leave it as it is even.
The risk is not that big perhaps but there's some. 

I will leave it up to you how to address it or not

Reply:
https://github.com/vmware/versatile-data-kit/issues/2337